smartsim/_core/launcher/dragon/dragonLauncher.py

```diff
@@ -368,13 +203,18 @@ def stop(self, step_name: str) -> StepInfo:
         :rtype: StepInfo
         """
 
-        if not self.is_connected:
-            raise LauncherError("Launcher is not connected to Dragon.")
-
         stepmap = self.step_mapping[step_name]
         step_id = str(stepmap.step_id)
+
+        if step_id.startswith("SLURM-"):
+            return self._slurm_launcher.stop(step_name)
+
+        if step_id.startswith("PBS-"):
```

Comment:
can we use `sublauncher.stop(step_name)` since sublauncher is already decided by this prefix?

Reply:
Understood. Maybe later we can consider an approach for inspecting that doesn't involve knowing that prefix, e.g. let the launcher look at the the step and decide so a change in prefix or methodology only has to be tied to the launcher

```
if self._some_launcher.can_run(step):
  ...
```